doc/cephfs/fs-volumes.rst

```diff
@@ -766,6 +766,362 @@ Will enable distributed subtree partitioning policy for the "csi" subvolume
 group.  This will cause every subvolume within the group to be automatically
 pinned to one of the available ranks on the file system.
 
+Subvolume quiesce
+-----------------
+
+It may be needed to pause (write) IO to a set of subvolumes of a given file system.
+A good example of such case is a consistent snapshot spanning multiple subvolumes.
+Such a task arises often in an environment such as k8s, where a single deployed application
+can work with many mounted subvolumes across several hosts. When a snapshot of such a system is needed,
+the application may not find the result consistent unless the snapshots were taken
+under an active write pause.
+
+The volumes plugin provides a tool to initiate and await such a pause across a set of subvolumes:
+
+.. prompt:: bash $ auto
+
+  $ ceph fs subvolume quiesce --set-id myset1 <vol_name> <[group_name/]sub_name...> --await
+  # perform actions while the IO pause is active, like taking snapshots
+  $ ceph fs subvolume quiesce --set-id myset1 --release --await
+  # if successful, all members of the set were confirmed as still in pause and released from such
+
+The ``subvolume quiesce`` functionality is itself based on top of a lower level QuiesceDb service maintained by the MDS
+daemons. Volumes plugin merely maps the subvolume names to their corresponding paths on the given file system
+and then issues the corresponding quiesce db command. You can learn more about the feature in the developer guides.
+
+Operations
+^^^^^^^^^^
+
+The IO pause (referred to as `quiesce`) is requested for a group of one or more subvolumes (i.e. paths in a filesystem).
+The group is referred to as "quiesce set", and every quiesce set must have a unique string id to interact with.
+A quiesce set can be manipulated in the following ways:
+
+* **include** one or more subvolumes - quiesce set members
+* **exclude** one or more members
+* **cancel** the set, asynchronously aborting the pause on all its current members
+* **release** the set, requesting the end of the pause from all members and expecting an ack from all clients
+* **query** the current state of a set by id or all active sets or all known sets
+* **cancel all** active sets in case an immediate resume of IO is required.
+* **await** the set to change states, applicable as a modifier to other requests
+
+All of the operations listed above are carried out syncronously by the designated quiesce leader,
```

Comment:
```suggestion
All of the operations listed above are carried out asynchronously by the designated quiesce leader,
```

?

Reply:
Probably not the best wording. I did intend to use `synchronously`, but I understand your logic here.

So, what I'm trying to say is that when `--await` or `--await-for` is not used, then the operations perform their designated changes and return with the latest version of the set they were targeted at. So, I can't call it `asynchronous` since they do pend on the quiesce db manager to apply them in the next batch. However, it shouldn't ever take long - the delays are just the transport and quiesce db manager control loop latency.

Reply:
Okay. In any case the spelling needs fixed :)

Reply:
That's very Pittsburghese of you ;)

Reply:
I do see how this was confusing for you though. I will appreciate suggestions to make it clearer.
I will probably just have to add more words, along the lines of my previous comment